File: abstract-factory/java/abstract-factory/src/main/java/ConcreteFactory/WindowsFactory.java
package ConcreteFactory;

// 5. Concrete Factories

import AbstractFactory.GUIFactory;
import AbstractProduct.Button;
import AbstractProduct.Checkbox;
import ConcreteProduct.WindowsButton;
import ConcreteProduct.WindowsCheckbox;

public class WindowsFactory implements GUIFactory {
    @Override
    public Button createButton() {
        return new WindowsButton();
    }

    @Override
    public Checkbox createCheckbox() {
        return new WindowsCheckbox();
    }
}

File: abstract-factory/go/abstract-factory-go/concrete-factory/WindowsFactory.go
package concretefactory

import (
	abstractproduct "github.com/OlyMahmudMugdho/design-patterns/abstract-factory/abstract-product"
	concreteproduct "github.com/OlyMahmudMugdho/design-patterns/abstract-factory/concrete-product"
)

type WindowsFactory struct{}

func NewWindowsFactory() *WindowsFactory {
	return &WindowsFactory{}
}

// WindowsButton implementation
func (wf *WindowsFactory) CreateButton() abstractproduct.Button {
	return new(concreteproduct.WindowsButton)
}
